File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_java_labs_and_solutions/template_method/lab/CashPurchase.java
// CashPurchase.java

public class CashPurchase extends Purchase
{
	public void perform()
	{
		selectItem();
		bringItemToRegister();
		payViaCash();
	}

	private void selectItem()
	{
	}

	private void bringItemToRegister()
	{
	}

	private void payViaCash()
	{
		System.out.println("Paying via cash.");
	}
}

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_matlab_labs_and_solutions/template_method/lab/CashPurchase.m
% CashPurchase.m

classdef CashPurchase < Purchase
	methods
		function perform(theCashPurchase)
			theCashPurchase.selectItem;
			theCashPurchase.bringItemToRegister;
			theCashPurchase.payViaCash;
		end
	end

	methods (Access = private)
		function selectItem(theCashPurchase)
		end

		function bringItemToRegister(theCashPurchase)
		end

		function payViaCash(theCashPurchase)
			fprintf('Paying via cash.\n');
		end
	end
end
